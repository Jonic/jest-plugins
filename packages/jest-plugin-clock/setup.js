/**
 * Installs clock and makes it available as a global function throughout all
 * test files.
 */
global.clock = require('./').default;
