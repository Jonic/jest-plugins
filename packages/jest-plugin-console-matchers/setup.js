const expect = require('jest-matchers');
const {
  toConsoleDebug,
  toConsoleError,
  toConsoleLog,
  toConsoleWarn,
  toThrowWarning,
} = require('./');


// Injects custom matchers into jest.
expect.extend({
  toConsoleDebug,
  toConsoleError,
  toConsoleLog,
  toConsoleWarn,
  toThrowWarning,
});
