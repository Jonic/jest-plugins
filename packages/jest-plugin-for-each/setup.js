/**
 * Installs forEach and makes it available as a global function throughout all
 * test files.
 */
global.forEach = require('./').default;
