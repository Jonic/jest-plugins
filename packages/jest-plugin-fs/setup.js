/**
 * Installs fs and makes it available on the jest global object throughout all
 * test files.
 */
jest.fs = require('./').default;
