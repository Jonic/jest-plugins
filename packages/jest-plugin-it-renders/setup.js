/**
 * Installs itRenders and makes it available as a global function throughout all
 * test files.
 */
global.itRenders = require('./').default;
