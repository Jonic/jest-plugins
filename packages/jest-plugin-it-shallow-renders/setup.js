/**
 * Installs itShallowRenders and makes it available as a global function
 * throughout all test files.
 */
global.itShallowRenders = require('./').default;
