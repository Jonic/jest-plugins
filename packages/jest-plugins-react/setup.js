// Sets up the plugins to help test React components.
require('jest-plugins')([
  'jest-plugin-it-renders',
]);
