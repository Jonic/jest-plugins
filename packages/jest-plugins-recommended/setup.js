// Sets up the recommended set of plugins.
require('jest-plugins')([
  'jest-plugins-rspec',

  'jest-plugin-action',
  'jest-plugin-console-matchers',
]);
