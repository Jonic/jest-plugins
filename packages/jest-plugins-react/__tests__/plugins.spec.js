// Modules
import '../setup';
import React from 'react';


// Test Components
const View = () => <div />;
const Text = () => <div />;
const Undefined = () => undefined;


/* eslint-disable no-undef */
describe('jest-plugins-react', () => {
  const globals = [
    'itRenders',
  ];

  globals.forEach(globalFunction => {
    describe(globalFunction, () => {
      it('should be defined as a global', () => {
        expect(global[globalFunction]).toBeDefined();
      });
    });
  });

  describe('itRenders', () => {
    itRenders(() => <View />);
    itRenders(() => <Text />);
  });
});
