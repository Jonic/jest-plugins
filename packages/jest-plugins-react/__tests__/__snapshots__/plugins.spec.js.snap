// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-plugins-react itRenders renders snapshot 1`] = `<div />`;

exports[`jest-plugins-react itRenders renders snapshot 2`] = `<div />`;

exports[`jest-plugins-react itShallowRenders shallow renders snapshot 1`] = `<div />`;

exports[`jest-plugins-react itShallowRenders shallow renders snapshot 2`] = `
<View>
  <div />
</View>
`;
